miner/env_changes.go

```diff
@@ -0,0 +1,417 @@
+
+		switch {
+		case errors.Is(err, core.ErrGasLimitReached):
+			// Pop the current out-of-gas transaction without shifting in the next from the account
+			from, _ := types.Sender(signer, tx)
+			log.Trace("Gas limit exceeded for current block", "sender", from)
+			return receipt, popTx, err
+
+		case errors.Is(err, core.ErrNonceTooLow):
+			// New head notification data race between the transaction pool and miner, shift
+			from, _ := types.Sender(signer, tx)
+			log.Trace("Skipping transaction with low nonce", "sender", from, "nonce", tx.Nonce())
+			return receipt, shiftTx, err
+
+		case errors.Is(err, core.ErrNonceTooHigh):
+			// Reorg notification data race between the transaction pool and miner, skip account =
+			from, _ := types.Sender(signer, tx)
+			log.Trace("Skipping account with hight nonce", "sender", from, "nonce", tx.Nonce())
+			return receipt, popTx, err
+
+		case errors.Is(err, core.ErrTxTypeNotSupported):
+			// Pop the unsupported transaction without shifting in the next from the account
+			from, _ := types.Sender(signer, tx)
+			log.Trace("Skipping unsupported transaction type", "sender", from, "type", tx.Type())
+			return receipt, popTx, err
+
+		default:
+			// Strange error, discard the transaction and get the next in line (note, the
+			// nonce-too-high clause will prevent us from executing in vain).
+			log.Trace("Transaction failed, account skipped", "hash", tx.Hash(), "err", err)
+			return receipt, shiftTx, err
+		}
+	}
+
+	for _, address := range touchTracer.TouchedAddresses() {
+		if _, in := chData.blacklist[address]; in {
+			c.rollback(usedGasBefore, gasPoolBefore, profitBefore, txsBefore, receiptsBefore)
+			return nil, popTx, errors.New("blacklist violation, tx trace")
+		}
+	}
+
+	c.profit.Add(c.profit, new(big.Int).Mul(new(big.Int).SetUint64(receipt.GasUsed), gasPrice))
+	c.txs = append(c.txs, tx)
+	c.receipts = append(c.receipts, receipt)
+
+	return receipt, shiftTx, nil
+}
+
+func (c *envChanges) commitBundle(bundle *types.SimulatedBundle, chData chainData) error {
+	var (
+		profitBefore   = new(big.Int).Set(c.profit)
+		coinbaseBefore = new(big.Int).Set(c.env.state.GetBalance(c.env.coinbase))
+		gasUsedBefore  = c.usedGas
+		gasPoolBefore  = new(core.GasPool).AddGas(c.gasPool.Gas())
+		txsBefore      = c.txs[:]
+		receiptsBefore = c.receipts[:]
+		hasBaseFee     = c.env.header.BaseFee != nil
+
+		bundleErr error
+	)
+
+	for _, tx := range bundle.OriginalBundle.Txs {
+		if hasBaseFee && tx.Type() == types.DynamicFeeTxType {
+			// Sanity check for extremely large numbers
+			if tx.GasFeeCap().BitLen() > 256 {
+				bundleErr = core.ErrFeeCapVeryHigh
+				break
+			}
+			if tx.GasTipCap().BitLen() > 256 {
+				bundleErr = core.ErrTipVeryHigh
+				break
+			}
+
+			// Ensure gasFeeCap is greater than or equal to gasTipCap.
+			if tx.GasFeeCapIntCmp(tx.GasTipCap()) < 0 {
+				bundleErr = core.ErrTipAboveFeeCap
+				break
+			}
+		}
+		receipt, _, err := c.commitTx(tx, chData)
+
+		if err != nil {
```

Comment:
what about reverting hashes feature that you implemented for diferent pr? or you decided to postpone it?

Reply:
we were waiting on feedback from searchers before merging that - I plan to deploy and merge that feature this week

Reply:
I've merged in support for reverting hashes feature to this branch